cumulus_library/studies/core/encounter.sql

```diff
@@ -30,87 +36,61 @@ SELECT DISTINCT
+-- ############################################################################
+-- # Encounter.Type, Encounter.serviceType, Encounter.Priority
 
-
-CREATE TABLE core__count_encounter_month AS
-WITH powerset AS (
-    SELECT
-        count(DISTINCT ce.subject_ref) AS cnt_subject,
-        count(DISTINCT ce.encounter_id) AS cnt_encounter,
-        ce.enc_class.code AS enc_class_code,
-        ce.start_month,
-        ce.age_at_visit,
-        cp.gender,
-        cp.race_display,
-        cp.ethnicity_display
-    FROM core__encounter AS ce, core__patient AS cp
-    WHERE ce.subject_ref = cp.subject_ref
-    GROUP BY
-        cube(
-            ce.enc_class,
-            ce.start_month,
-            ce.age_at_visit,
-            cp.gender,
-            cp.race_display,
-            cp.ethnicity_display
-        )
-)
-
-SELECT DISTINCT
-    powerset.cnt_encounter AS cnt,
-    powerset.enc_class_code,
-    powerset.start_month,
-    powerset.age_at_visit,
-    powerset.gender,
-    powerset.race_display,
-    powerset.ethnicity_display
-FROM powerset
-WHERE powerset.cnt_subject >= 10
-ORDER BY
-    powerset.start_month ASC, powerset.enc_class_code ASC, powerset.age_at_visit ASC;
-
-CREATE TABLE core__count_encounter_day AS
-WITH powerset AS (
-    SELECT
-        count(DISTINCT ce.subject_ref) AS cnt_subject,
-        count(DISTINCT ce.encounter_id) AS cnt_encounter,
-        ce.enc_class.code AS enc_class_code,
-        ce.start_date
-    FROM core__encounter AS ce, core__patient AS cp
-    WHERE ce.subject_ref = cp.subject_ref
-    GROUP BY cube(ce.enc_class, ce.start_date)
+create table core__encounter_type as WITH
+join_enc_type as (
+    select distinct encounter_id, as_coding from core__encounter,
+    unnest(enc_type) as t (as_row),
+    unnest(as_row.coding) as t (as_coding)
+),
+join_service as (
+    select distinct encounter_id, as_coding from core__encounter,
+    unnest(service_type.coding) as t (as_coding)
+),
+join_priority as (
+    select distinct encounter_id, as_coding from core__encounter,
+    unnest(priority.coding) as t (as_coding)
 )
-
 SELECT DISTINCT
-    cnt_encounter AS cnt,
-    enc_class_code,
-    start_date
-FROM powerset
-WHERE cnt_subject >= 10
-ORDER BY start_date ASC, enc_class_code ASC;
+    E.enc_class_code,
+    E.enc_class_display,
+    COALESCE(join_enc_type.as_coding.system, '?')   as enc_type_system,
+    COALESCE(join_enc_type.as_coding.code, '?')     as enc_type_code,
+    COALESCE(join_enc_type.as_coding.display, '?')  as enc_type_display,
+    COALESCE(join_service.as_coding.system, '?')    as enc_service_system,
+    COALESCE(join_service.as_coding.code, '?')      as enc_service_code,
+    COALESCE(join_service.as_coding.display, '?')   as enc_service_display,
+    COALESCE(join_priority.as_coding.system, '?')   as enc_priority_system,
+    COALESCE(join_priority.as_coding.code, '?')     as enc_priority_code,
+    COALESCE(lower(join_priority.as_coding.display), '?')  as enc_priority_display,
```

Comment:
I'm leery of not just using nulls for stuff like this - it's the expected behavior, so it adds an unusual condition if you're trying to check for presence/absence later in a downstream query against this table.

Reply:
it is intentional. We only sometimes get Encounter type, service, priority.

Reply:
was resolved via #75